src/ppx/scene/scene_node.cpp

```diff
@@ -0,0 +1,250 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ppx/scene/scene_node.h"
+
+namespace ppx {
+namespace scene {
+
+// -------------------------------------------------------------------------------------------------
+// Node
+// -------------------------------------------------------------------------------------------------
+Node::Node(scene::Scene* pScene)
+    : mScene(pScene)
+{
+}
+
+Node::~Node()
+{
+}
+
+void Node::SetVisible(bool visible)
+{
+    mVisible = visible;
```

Comment:
Does this need to set visibility of the children or no?

Reply:
Good question. Generally it's left up to the app on what to do. Added parameter to recurse if desired.